### PR TITLE
Use ubuntu 2024.04 in CI

### PR DESCRIPTION
**Pull request by rasolca**

### CODE REVIEW

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

File: test/unit/c_api/grid/test_grid_c_api.cpp [hunk 1]
```diff
@@ -43,7 +43,7 @@ TEST(GridTest, GridScaLAPACKOrdering) {
 #endif
 
 TEST(GridTest, GridDLAFOrdering) {
-  for (const auto [key, value] : ordering) {
+  for (const auto& [key, value] : ordering) {
```
Comment (msimberg) [inline test/unit/c_api/grid/test_grid_c_api.cpp:46]: Just curious, was this a warning or compilation error? Did we miss testing something previously?

**Review by msimberg — APPROVED** [review]

File: test/unit/c_api/grid/test_grid_c_api.cpp (hunk 1 above)
Comment (rasolca) [reply test/unit/c_api/grid/test_grid_c_api.cpp:46]: A new warning, IIRC, with clang 18

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Comment by rasolca** [comment]
cscs-ci run

**Pull request merged**